src/python/pants/engine/unions.py

```diff
@@ -59,6 +61,12 @@ def __init__(self, union_rules: Mapping[Type, Iterable[Type]]) -> None:
             {base: FrozenOrderedSet(members) for base, members in union_rules.items()}
         )
 
+    def __getitem__(self, union_type: Type[_T]) -> FrozenOrderedSet[Type[_T]]:
```

Comment:
It might be good to include some guidance here on when to use each of these.

The `__getitem__` variant should probably only be used when a ruleset is both requesting and providing the union... presumably because they want to allow for extension beyond some default, and `get` would be recommended in more cases?

But it looks like we use `__getitem__` in more places based on an assertion that the goal won't run if there aren't any backends installed... it feels like we might want to enrich the error message a bit in case that assertion is ever broken...?

Reply:
I'll add docstring. The only reason to use `__getitem__` is that you're confident the key will be in the dictionary, as it's less typing.